src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,45 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final Integer MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
+    public boolean existLogin(String login) {
+        for (User person : Storage.people) {
+            if (person.getLogin().equals(login)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
+        } else if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("Login must be longer than 6 characters");
+        } else if (existLogin(user.getLogin())) {
+            throw new RegistrationException(user.getLogin() + " has already been taken");
```

Comment:
Consider adding a space before 'has already been taken' for better readability in the error message.